pkg/db/db_test.go

```diff
@@ -1823,6 +1837,1151 @@ func TestReadReleasesByApp(t *testing.T) {
+									"test": {
+										Name:    "test",
+										Version: 1,
+										DeploymentMetaData: &api.Environment_Application_DeploymentMetaData{
+											DeployAuthor: "testmail@example.com",
+											DeployTime:   "1",
+										},
+										Team: "team-123",
+									},
+								},
+								Priority: api.Priority_YOLO,
+							},
+						},
+						Priority: api.Priority_YOLO,
+					},
+				},
+				Applications: map[string]*api.Application{
+					"test": {
+						Name: "test",
+						Releases: []*api.Release{
+							{
+								Version:        1,
+								SourceCommitId: "changedcommitId",
+								SourceAuthor:   "changedAuthor",
+								SourceMessage:  "changed changed something (#679)",
+								PrNumber:       "679",
+								CreatedAt:      &timestamppb.Timestamp{Seconds: 1, Nanos: 1},
+							},
+						},
+						Team: "team-123",
+					},
+				},
+				GitRevision: "0",
+			},
+		},
+
+		{
+			Name: "app does not exists",
+			NewRelease: DBReleaseWithMetaData{
+				App:           "does-not-exists",
+				ReleaseNumber: 12,
+				Metadata: DBReleaseMetaData{
+					SourceAuthor:   "testmail@example.com",
+					SourceCommitId: "testcommit",
+					SourceMessage:  "changed something (#677)",
+					DisplayVersion: "12",
+				},
+				Created: time.Date(2024, time.July, 12, 15, 30, 0, 0, time.UTC),
+			},
+			ExpectedError: errMatcher{"could not find application does-not-exists in overview"},
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.Name, func(t *testing.T) {
+			t.Parallel()
+			ctx := testutil.MakeTestContext()
+			dbHandler := setupDB(t)
+
+			err := dbHandler.WithTransaction(ctx, false, func(ctx context.Context, transaction *sql.Tx) error {
+				err := dbHandler.WriteOverviewCache(ctx, transaction, startingOverview)
+				if err != nil {
+					return err
+				}
+				err = dbHandler.UpdateOverviewRelease(ctx, transaction, tc.NewRelease)
+				if err != nil {
+					if diff := cmp.Diff(tc.ExpectedError, err, cmpopts.EquateErrors()); diff != "" {
+						return fmt.Errorf("mismatch between errors (-want +got):\n%s", diff)
+					}
+					return nil
+				}
+				latestOverview, err := dbHandler.ReadLatestOverviewCache(ctx, transaction)
+				if err != nil {
+					return err
+				}
+				opts := cmpopts.IgnoreUnexported(api.GetOverviewResponse{}, api.EnvironmentGroup{}, api.Environment{}, api.Application{}, api.Release{}, timestamppb.Timestamp{}, api.EnvironmentConfig{}, api.EnvironmentConfig_Upstream{}, api.Environment_Application{}, api.Environment_Application_DeploymentMetaData{}, api.EnvironmentConfig_ArgoCD{}, api.Lock{}, api.Actor{})
+				if diff := cmp.Diff(tc.ExcpectedOverview, latestOverview, opts); diff != "" {
+					return fmt.Errorf("mismatch (-want +got):\n%s", diff)
+				}
+				return nil
+			})
+
+			if err != nil {
+				t.Fatal(err)
+			}
+		})
+	}
+}
+
+func TestForceOverviewRecalculation(t *testing.T) {
+	t.Run("ForceOverviewRecalculation", func(t *testing.T) {
```

Comment:
even if you don't need the table-driven approach right now, please make it easier for the next person who wants to add another test.
=> Add the test array with one element and iterate over it, like in other tests.